### ISSUE TITLE
[0.12.5] broken build

### ISSUE THREAD

**Issue by remicollet** [zephir-lang/zephir#1940]
```
/bin/sh /dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/libtool --mode=compile cc  -I. -I/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext -DPHP_ATOM_INC -I/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/include -I/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/main -I/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection   -c /dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/kernel/require.c -o kernel/require.lo 
In file included from /dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/kernel/string.c:51:
/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/kernel/string.c: In function 'zephir_preg_match':
./kernel/memory.h:82:66: error: 'ZEPHIR_METHOD_GLOBALS_PTR' undeclared (first use in this function); did you mean 'ZEPHIR_KERNEL_GLOBALS_H'?
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/memory.h:94:2: note: in expansion of macro 'zephir_memory_observe'
  zephir_memory_observe(z); \
  ^~~~~~~~~~~~~~~~~~~~~
/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/kernel/string.c:1117:2: note: in expansion of macro 'ZEPHIR_INIT_VAR'
  ZEPHIR_INIT_VAR(&tmp_flags);
  ^~~~~~~~~~~~~~~
./kernel/memory.h:82:66: note: each undeclared identifier is reported only once for each function it appears in
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
./kernel/memory.h:94:2: note: in expansion of macro 'zephir_memory_observe'
  zephir_memory_observe(z); \
  ^~~~~~~~~~~~~~~~~~~~~
/dev/shm/BUILD/zephir-0f4dfe0398dd11035c729b38b4d9a6a0a52263fc/ext/kernel/string.c:1117:2: note: in expansion of macro 'ZEPHIR_INIT_VAR'
  ZEPHIR_INIT_VAR(&tmp_flags);
  ^~~~~~~~~~~~~~~
make[1]: *** [Makefile:210: kernel/string.lo] Error 1

```

**Comment by ruudboon** [comment]
Seeing something similar in Nanobox with Phalcon
```
make: Warning: File 'Makefile' has modification time 89 s in the future
In file included from /app/ext/kernel/string.c:51:0:
/app/ext/kernel/string.c: In function ‘zephir_json_encode’:
./kernel/memory.h:82:66: error: ‘ZEPHIR_METHOD_GLOBALS_PTR’ undeclared (first use in this function)
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1194:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zopts);
  ^
./kernel/memory.h:82:66: note: each undeclared identifier is reported only once for each function it appears in
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1194:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zopts);
  ^
/app/ext/kernel/string.c: In function ‘zephir_json_decode’:
./kernel/memory.h:82:66: error: ‘ZEPHIR_METHOD_GLOBALS_PTR’ undeclared (first use in this function)
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1208:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zassoc);
  ^
make: *** [kernel/string.lo] Error 1
make: *** Waiting for unfinished jobs....
make: Warning: File 'Makefile' has modification time 87 s in the future
In file included from /app/ext/kernel/string.c:51:0:
/app/ext/kernel/string.c: In function ‘zephir_json_encode’:
./kernel/memory.h:82:66: error: ‘ZEPHIR_METHOD_GLOBALS_PTR’ undeclared (first use in this function)
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1194:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zopts);
  ^
./kernel/memory.h:82:66: note: each undeclared identifier is reported only once for each function it appears in
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1194:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zopts);
  ^
/app/ext/kernel/string.c: In function ‘zephir_json_decode’:
./kernel/memory.h:82:66: error: ‘ZEPHIR_METHOD_GLOBALS_PTR’ undeclared (first use in this function)
 #define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);
                                                                  ^
./kernel/memory.h:94:2: note: in expansion of macro ‘zephir_memory_observe’
  zephir_memory_observe(z); \
  ^
/app/ext/kernel/string.c:1208:2: note: in expansion of macro ‘ZEPHIR_INIT_VAR’
  ZEPHIR_INIT_VAR(&zassoc);
  ^
make: *** [kernel/string.lo] Error 1
```

**Comment by sergeyklay** [comment]
@remicollet Does adding `#include "kernel/memory.h"` to the `string.c` solves the issue?

**Comment by remicollet** [comment]
@sergeyklay it is already included.

**Comment by sergeyklay** [comment]
@remicollet I'll fix this ASAP. Could you provide steps to reproduce this issue? Unfortunately,  tests are passed successfully, so I didn't known about this problem

**Comment by ruudboon** [comment]
I have a branch ready:
```
git clone git@github.com:ruudboon/cphalcon.git -b zephir-upgrade zephir-upgrade
cd zephir-upgrade
cp -v ./tests/_ci/nanobox/boxfile.7.3.yml ./boxfile.yml
nanobox run
zephir fullclean
zephir build
cat compile-errors.log
```

**Comment by sergeyklay** [comment]
@ruudboon In general I need reproducible steps I can add to current .travis.yml

**Comment by ruudboon** [comment]
@remicollet Do you have some more information about your environment? Are you also using nanobox?

**Comment by sergeyklay** [comment]
@remicollet It seems I found the root of problem. Do you have ext-json installed/enabled? 
Is this check works for you
https://github.com/phalcon/zephir/blob/99f41ba084c5c454fda9421bea77eaf904c8ba3e/templates/ZendEngine3/config.m4#L36-L51

---

@dreamsxin 

> ./kernel/memory.h:82:66: error
> ./kernel/memory.h:94:2: note
>  /app/ext/kernel/string.c:1194:2: note

Take a look at zephir_json_decode:

```c
#ifdef ZEPHIR_USE_PHP_JSON
// Some code here
// ...
/// ...
#else

int zephir_json_encode(zval *return_value, zval *v, int opts)
{
	zval zopts;
	zval *params[2];

	ZEPHIR_INIT_VAR(&zopts);; // <------ Line 1194 (previously ZEPHIR_SINIT_VAR(zopts))
	ZVAL_LONG(&zopts, opts);

	params[0] = v;
	params[1] = &zopts;

	return zephir_return_call_function(return_value, SL("json_encode"), NULL, 0, 2, params);
}

// Some code here
// ...
/// ...

#endif /* ZEPHIR_USE_PHP_JSON */
```

Then take a look at `memory.h`:
```c
#define zephir_memory_observe(var) zephir_do_memory_observe(var, ZEPHIR_METHOD_GLOBALS_PTR);

#define ZEPHIR_INIT_VAR(z) \
	zephir_memory_observe(z); \  // <----- ./kernel/memory.h:94:2: note
	ZVAL_NULL(z);
```

Looks like @remicollet and @ruudboon uses `zephir_memory_observe` with `ZEPHIR_METHOD_GLOBALS_PTR` variable. But this variable should be set through `ZEPHIR_MM_GROW()` macro. So somewhere missed a call of the `ZEPHIR_MM_GROW()` macro

**Comment by ruudboon** [comment]
@dreamsxin I used `git bisect` to traceback the commit. Outcome for me was this commit https://github.com/phalcon/zephir/commit/0922dad6ef9c8178644f78fe15b0137baf74d25b

Hopefully it can be at help. Thank you.

**Comment by sergeyklay** [comment]
@ruudboon @remicollet I need your config.h. Could you please share the content

**Comment by ruudboon** [comment]
ext\config.h
```
/* config.h.  Generated from config.h.in by configure.  */
/* config.h.in.  Generated from configure.ac by autoheader.  */

/* Whether to build phalcon as dynamic module */
#define COMPILE_DL_PHALCON 1

/* Define to 1 if you have the <dlfcn.h> header file. */
#define HAVE_DLFCN_H 1

/* Define to 1 if you have the <ext/json/php_json.h> header file. */
/* #undef HAVE_EXT_JSON_PHP_JSON_H */

/* Define to 1 if you have the <ext/pcre/php_pcre.h> header file. */
#define HAVE_EXT_PCRE_PHP_PCRE_H 1

/* Define to 1 if you have the <inttypes.h> header file. */
#define HAVE_INTTYPES_H 1

/* Define to 1 if you have the <memory.h> header file. */
#define HAVE_MEMORY_H 1

/* Whether you have Phalcon */
#define HAVE_PHALCON 1

/* Define to 1 if you have the <stdint.h> header file. */
#define HAVE_STDINT_H 1

/* Define to 1 if you have the <stdlib.h> header file. */
#define HAVE_STDLIB_H 1

/* Define to 1 if you have the <strings.h> header file. */
#define HAVE_STRINGS_H 1

/* Define to 1 if you have the <string.h> header file. */
#define HAVE_STRING_H 1

/* Define to 1 if you have the <sys/stat.h> header file. */
#define HAVE_SYS_STAT_H 1

/* Define to 1 if you have the <sys/types.h> header file. */
#define HAVE_SYS_TYPES_H 1

/* Define to 1 if you have the <unistd.h> header file. */
#define HAVE_UNISTD_H 1

/* Define to 1 if your C compiler doesn't accept -c and -o together. */
/* #undef NO_MINUS_C_MINUS_O */

/* Define to the address where bug reports for this package should be sent. */
#define PACKAGE_BUGREPORT ""

/* Define to the full name of this package. */
#define PACKAGE_NAME ""

/* Define to the full name and version of this package. */
#define PACKAGE_STRING ""

/* Define to the one symbol short name of this package. */
#define PACKAGE_TARNAME ""

/* Define to the home page for this package. */
#define PACKAGE_URL ""

/* Define to the version of this package. */
#define PACKAGE_VERSION ""

/* Define to 1 if you have the ANSI C header files. */
#define STDC_HEADERS 1

/* Whether PHP json extension is present at compile time */
/* #undef ZEPHIR_USE_PHP_JSON */

/* Whether PHP pcre extension is present at compile time */
#define ZEPHIR_USE_PHP_PCRE 1
```

**Comment by sergeyklay** [comment]
@dreamsxin Do you see ^ 
```c
/* Whether PHP json extension is present at compile time */
/* #undef ZEPHIR_USE_PHP_JSON */
```

**Comment by sergeyklay** [comment]
The issue was in:
```bash
$ grep -c '#define HAVE_JSON 1' $(phpenv prefix)/include/php/main/php_config.h
1
```

I'm pretty sure @remicollet and @ruudboon does not have HAVE_JSON constant or it set to 0

**Comment by sergeyklay** [comment]
Fixed in the `development` branch. Feel free to open a new issue if the problem appears again. Thank you for the bug report.

**Comment by remicollet** [comment]
Still encounter the same issue with 0.12.6

Also "zephir api" raise an error
`Parse error: syntax error, unexpected 'version' (T_STRING) in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/templates/Api/sitemap.php on line 1
`

Full output
```
$ php  -d memory_limit=1G -d extension=zephir_parser zephir.php clean
$ php  -d memory_limit=1G -d extension=zephir_parser zephir.php fullclean
$ php  -d memory_limit=1G -d extension=zephir_parser zephir.php generate
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 662 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 679 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "elements" assigned but not used in Test\Assign::testConstantKeyAssign in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 898 [unused-variable]

	  let elements = ["abc": 1, ABDAY_1: DAY_1, ABDAY_2: DAY_2];

	---------------------------------------------------------^
 Warning: Variable "abc" assigned but not used in Test\Closures::testUseCommand in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/closures.zep on line 54 [unused-variable]

	  var abc = 1;

	-------------^
 Warning: Constant 'ANNOTATION_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 45 [nonexistent-constant]

	 protected propertyC7 = self::ANNOTATION_REGEX {get};

	-----------------------------------------------^
 Warning: Constant 'PARAMETER_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 46 [nonexistent-constant]

	 protected propertyC8 = self::PARAMETER_REGEX {get};

	----------------------------------------------^
 Warning: Magic constant '__DIR__' is not supported in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 105 [not-supported-magic-constant]

	  return __DIR__;

	----------------^
 Warning: Variable "e" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 135 [unused-variable]

	  return 1;

	-------^
 Warning: Variable "status" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 132 [unused-variable]

	   let status = "woop";

	--------------------^
 Warning: Unreachable code in Test\Fannkuch::process in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fannkuch.zep on line 36 [unreachable-code]

	   }

	---^
 Warning: Variable "seqi" assigned but not used in Test\Fasta::fastaRepeat in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 9 [unused-variable]

	  var seqi = 0, len, i, j, k, l, block, str, lines;

	--------------^
 Warning: Variable "iub" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 65 [unused-variable]

	     "M": 0.02, "N": 0.02, "R": 0.02, "S": 0.02, "V": 0.02, "W": 0.02, "Y": 0.02 ];

	--------------------------------------------------------------------^
 Warning: Variable "homoSap" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 69 [unused-variable]

	  ];

	---^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 16 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray3 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 27 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray5 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 38 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchObject1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 49 [unused-variable]

	  return fetch c, a->{b};

	------------------------^
 Warning: Unreachable code in Test\Flow::testIf2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 26 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 46 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 66 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 78 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf8 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 87 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 130 [unreachable-code]

	            return -3;

	---------------------^
 Warning: Unreachable code in Test\Flow::testIf12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 153 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 394 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 404 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Variable "v" assigned but not used in Test\Flow::testFor2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 450 [unused-variable]

	        for k, v in b {

	----------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 795 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 799 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 803 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 823 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 829 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 835 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Constant 'MYSQL_ASSOC' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow/switchflow.zep on line 171 [nonexistent-constant]

	            case MYSQL_ASSOC:

	----------------------------^
 Warning: Variable "c" assigned but not used in Test\NativeArray::testMultipleArrayUpdate13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/nativearray.zep on line 538 [unused-variable]

	   c = b,

	--------^
 Warning: Variable "a" assigned but not used in Test\Oo\DynamicProp::setPropertyVariableString in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/dynamicprop.zep on line 56 [unused-variable]

	  var a = "test";

	--------------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 93 [nonexistent-class]

	      );

	-------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 98 [nonexistent-class]

	     );

	------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 107 [nonexistent-class]

	    );

	-----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 146 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 154 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 171 [nonexistent-class]

	   );

	----^
 Warning: Unreachable code in Test\Returns::testReturnCast4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/returns.zep on line 26 [unreachable-code]

	   return (int)0;

	----------------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 90 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 106 [unused-variable]

	 }

	-^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry9 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 138 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 153 [unused-variable]

	  return false;

	-------^
 Warning: Variable "ex" assigned but not used in Test\TryTest::testTry11 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 166 [unused-variable]

	    }

	----^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 12 [nonexistent-function]

	  return new string(24);

	-----------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 17 [nonexistent-function]

	  return new string(-24);

	------------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 22 [nonexistent-function]

	  return new string(0);

	----------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 26 [nonexistent-class]

	  return new OoConstruct();

	--------------------------^
 Warning: Class "Strings" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 31 [nonexistent-class]

	  return new TestStrings();

	--------------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 36 [nonexistent-class]

	  return new NsAlias\OoConstruct();

	----------------------------------^
 Warning: Variable "a" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 114 [unused-variable]

	        int a = 1;

	-----------------^
 Warning: Variable "pi" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 115 [unused-variable]

	        double pi = 3.14;

	------------------------^
$ php  -d memory_limit=1G -d extension=zephir_parser zephir.php stubs
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 662 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 679 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "elements" assigned but not used in Test\Assign::testConstantKeyAssign in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 898 [unused-variable]

	  let elements = ["abc": 1, ABDAY_1: DAY_1, ABDAY_2: DAY_2];

	---------------------------------------------------------^
 Warning: Variable "abc" assigned but not used in Test\Closures::testUseCommand in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/closures.zep on line 54 [unused-variable]

	  var abc = 1;

	-------------^
 Warning: Constant 'ANNOTATION_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 45 [nonexistent-constant]

	 protected propertyC7 = self::ANNOTATION_REGEX {get};

	-----------------------------------------------^
 Warning: Constant 'PARAMETER_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 46 [nonexistent-constant]

	 protected propertyC8 = self::PARAMETER_REGEX {get};

	----------------------------------------------^
 Warning: Magic constant '__DIR__' is not supported in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 105 [not-supported-magic-constant]

	  return __DIR__;

	----------------^
 Warning: Variable "e" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 135 [unused-variable]

	  return 1;

	-------^
 Warning: Variable "status" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 132 [unused-variable]

	   let status = "woop";

	--------------------^
 Warning: Unreachable code in Test\Fannkuch::process in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fannkuch.zep on line 36 [unreachable-code]

	   }

	---^
 Warning: Variable "seqi" assigned but not used in Test\Fasta::fastaRepeat in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 9 [unused-variable]

	  var seqi = 0, len, i, j, k, l, block, str, lines;

	--------------^
 Warning: Variable "iub" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 65 [unused-variable]

	     "M": 0.02, "N": 0.02, "R": 0.02, "S": 0.02, "V": 0.02, "W": 0.02, "Y": 0.02 ];

	--------------------------------------------------------------------^
 Warning: Variable "homoSap" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 69 [unused-variable]

	  ];

	---^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 16 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray3 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 27 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray5 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 38 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchObject1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 49 [unused-variable]

	  return fetch c, a->{b};

	------------------------^
 Warning: Unreachable code in Test\Flow::testIf2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 26 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 46 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 66 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 78 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf8 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 87 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 130 [unreachable-code]

	            return -3;

	---------------------^
 Warning: Unreachable code in Test\Flow::testIf12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 153 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 394 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 404 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Variable "v" assigned but not used in Test\Flow::testFor2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 450 [unused-variable]

	        for k, v in b {

	----------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 795 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 799 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 803 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 823 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 829 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 835 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Constant 'MYSQL_ASSOC' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow/switchflow.zep on line 171 [nonexistent-constant]

	            case MYSQL_ASSOC:

	----------------------------^
 Warning: Variable "c" assigned but not used in Test\NativeArray::testMultipleArrayUpdate13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/nativearray.zep on line 538 [unused-variable]

	   c = b,

	--------^
 Warning: Variable "a" assigned but not used in Test\Oo\DynamicProp::setPropertyVariableString in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/dynamicprop.zep on line 56 [unused-variable]

	  var a = "test";

	--------------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 93 [nonexistent-class]

	      );

	-------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 98 [nonexistent-class]

	     );

	------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 107 [nonexistent-class]

	    );

	-----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 146 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 154 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 171 [nonexistent-class]

	   );

	----^
 Warning: Unreachable code in Test\Returns::testReturnCast4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/returns.zep on line 26 [unreachable-code]

	   return (int)0;

	----------------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 90 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 106 [unused-variable]

	 }

	-^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry9 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 138 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 153 [unused-variable]

	  return false;

	-------^
 Warning: Variable "ex" assigned but not used in Test\TryTest::testTry11 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 166 [unused-variable]

	    }

	----^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 12 [nonexistent-function]

	  return new string(24);

	-----------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 17 [nonexistent-function]

	  return new string(-24);

	------------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 22 [nonexistent-function]

	  return new string(0);

	----------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 26 [nonexistent-class]

	  return new OoConstruct();

	--------------------------^
 Warning: Class "Strings" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 31 [nonexistent-class]

	  return new TestStrings();

	--------------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 36 [nonexistent-class]

	  return new NsAlias\OoConstruct();

	----------------------------------^
 Warning: Variable "a" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 114 [unused-variable]

	        int a = 1;

	-----------------^
 Warning: Variable "pi" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 115 [unused-variable]

	        double pi = 3.14;

	------------------------^
 Generating stubs...
$ php  -d memory_limit=1G -d extension=zephir_parser zephir.php api
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 662 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "temp3" assigned but not used in Test\Assign::testPropertyArray13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 679 [unused-variable]

	  let temp3 = count(this->myArray);

	----------------------------------^
 Warning: Variable "elements" assigned but not used in Test\Assign::testConstantKeyAssign in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/assign.zep on line 898 [unused-variable]

	  let elements = ["abc": 1, ABDAY_1: DAY_1, ABDAY_2: DAY_2];

	---------------------------------------------------------^
 Warning: Variable "abc" assigned but not used in Test\Closures::testUseCommand in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/closures.zep on line 54 [unused-variable]

	  var abc = 1;

	-------------^
 Warning: Constant 'ANNOTATION_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 45 [nonexistent-constant]

	 protected propertyC7 = self::ANNOTATION_REGEX {get};

	-----------------------------------------------^
 Warning: Constant 'PARAMETER_REGEX' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 46 [nonexistent-constant]

	 protected propertyC8 = self::PARAMETER_REGEX {get};

	----------------------------------------------^
 Warning: Magic constant '__DIR__' is not supported in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/constants.zep on line 105 [not-supported-magic-constant]

	  return __DIR__;

	----------------^
 Warning: Variable "e" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 135 [unused-variable]

	  return 1;

	-------^
 Warning: Variable "status" assigned but not used in Test\Exceptions::issue1325 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/exceptions.zep on line 132 [unused-variable]

	   let status = "woop";

	--------------------^
 Warning: Unreachable code in Test\Fannkuch::process in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fannkuch.zep on line 36 [unreachable-code]

	   }

	---^
 Warning: Variable "seqi" assigned but not used in Test\Fasta::fastaRepeat in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 9 [unused-variable]

	  var seqi = 0, len, i, j, k, l, block, str, lines;

	--------------^
 Warning: Variable "iub" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 65 [unused-variable]

	     "M": 0.02, "N": 0.02, "R": 0.02, "S": 0.02, "V": 0.02, "W": 0.02, "Y": 0.02 ];

	--------------------------------------------------------------------^
 Warning: Variable "homoSap" assigned but not used in Test\Fasta::main in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fasta.zep on line 69 [unused-variable]

	  ];

	---^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 16 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray3 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 27 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchArray5 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 38 [unused-variable]

	  return fetch c, a[b];

	----------------------^
 Warning: Variable "c" assigned but not used in Test\FetchTest::testFetchObject1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/fetchtest.zep on line 49 [unused-variable]

	  return fetch c, a->{b};

	------------------------^
 Warning: Unreachable code in Test\Flow::testIf2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 26 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 46 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 66 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 78 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf8 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 87 [unreachable-code]

	            return 1;

	--------------------^
 Warning: Unreachable code in Test\Flow::testIf10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 130 [unreachable-code]

	            return -3;

	---------------------^
 Warning: Unreachable code in Test\Flow::testIf12 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 153 [unreachable-code]

	            return 0;

	--------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 394 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Possible attempt to decrement non-numeric dynamic variable in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 404 [non-valid-decrement]

	            let b--;

	-------------------^
 Warning: Variable "v" assigned but not used in Test\Flow::testFor2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 450 [unused-variable]

	        for k, v in b {

	----------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 795 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 799 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable1 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 803 [unreachable-code]

	            echo "hello true";

	---------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 823 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 829 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Unreachable code in Test\Flow::testUnrechable2 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow.zep on line 835 [unreachable-code]

	            echo "hello false";

	----------------------------^
 Warning: Constant 'MYSQL_ASSOC' does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/flow/switchflow.zep on line 171 [nonexistent-constant]

	            case MYSQL_ASSOC:

	----------------------------^
 Warning: Variable "c" assigned but not used in Test\NativeArray::testMultipleArrayUpdate13 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/nativearray.zep on line 538 [unused-variable]

	   c = b,

	--------^
 Warning: Variable "a" assigned but not used in Test\Oo\DynamicProp::setPropertyVariableString in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/dynamicprop.zep on line 56 [unused-variable]

	  var a = "test";

	--------------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 93 [nonexistent-class]

	      );

	-------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 98 [nonexistent-class]

	     );

	------^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 107 [nonexistent-class]

	    );

	-----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 146 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 154 [nonexistent-class]

	   );

	----^
 Warning: Class "Test\Oo\Exception" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/oo/oodestruct.zep on line 171 [nonexistent-class]

	   );

	----^
 Warning: Unreachable code in Test\Returns::testReturnCast4 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/returns.zep on line 26 [unreachable-code]

	   return (int)0;

	----------------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry6 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 90 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry7 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 106 [unused-variable]

	 }

	-^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry9 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 138 [unused-variable]

	  return false;

	-------^
 Warning: Variable "e" assigned but not used in Test\TryTest::testTry10 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 153 [unused-variable]

	  return false;

	-------^
 Warning: Variable "ex" assigned but not used in Test\TryTest::testTry11 in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/trytest.zep on line 166 [unused-variable]

	    }

	----^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 12 [nonexistent-function]

	  return new string(24);

	-----------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 17 [nonexistent-function]

	  return new string(-24);

	------------------------^
 Warning: Function "create_string" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/typeinstances.zep on line 22 [nonexistent-function]

	  return new string(0);

	----------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 26 [nonexistent-class]

	  return new OoConstruct();

	--------------------------^
 Warning: Class "Strings" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 31 [nonexistent-class]

	  return new TestStrings();

	--------------------------^
 Warning: Class "Oo\OoConstruct" does not exist at compile time in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/usetest.zep on line 36 [nonexistent-class]

	  return new NsAlias\OoConstruct();

	----------------------------------^
 Warning: Variable "a" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 114 [unused-variable]

	        int a = 1;

	-----------------^
 Warning: Variable "pi" assigned but not used in Test\Vars::testGetDefinedVars in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/test/vars.zep on line 115 [unused-variable]

	        double pi = 3.14;

	------------------------^
 Generating API into /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/doc/1.0.0

Parse error: syntax error, unexpected 'version' (T_STRING) in /dev/shm/BUILD/zephir-dd5e3f852ef284b700c02264ca6dcb21914e626c/templates/Api/sitemap.php on line 1

```

**Comment by remicollet** [comment]
Notice config.h have
```
/* Define to 1 if you have the <ext/pcre/php_pcre.h> header file. */
/* #undef HAVE_EXT_PCRE_PHP_PCRE_H */
```

Which is obviously wrong.
`ext/pcre/php_pcre.h` is there even if php is build with system pcre

**Comment by sergeyklay** [comment]
@remicollet I've fixed this right now. Could you please try development branch by any chance? Note, you'll need clone this repo, switch to the `development` branch and run `composer install`. Finally, use `./zephir` from the project root.

**Comment by remicollet** [comment]
I confirm using master at  98c5a73576e06c73886b7a4eb2bc69dfeafb3840
the extension build fine.

**Comment by sergeyklay** [comment]
@remicollet Just released 0.12.7 version: https://github.com/phalcon/zephir/releases/tag/0.12.7
Closing, this is fixed. If there are bugs, we can follow up in separate issues. Thank you for the bug report.
